app/controllers/media_objects_controller.rb

```diff
@@ -54,6 +55,26 @@ def confirm_remove
     raise CanCan::AccessDenied unless Array(params[:id]).any? { |id| current_ability.can? :destroy, MediaObject.find(id) }
   end
 
+  def intercom_collections
+    respond_to do |format|
+      format.json do
+        render json: Avalon::Intercom.user_collections(user_key).to_json
+      end
+    end
+  end
+
+  def intercom_push
+    result = Avalon::Intercom.push_media_object(@media_object, params[:collection_id], params[:include_structure] == 'true')
+    if result['id'].present?
+      target_url = URI.join(Avalon::Intercom.avalons['default']['url'], 'media_objects/', result['id'])
+      target_link = link_to "See it here", target_url, target: "_blank"
+      flash[:success] = "The item was pushed successfully. #{target_link}"
+    else
+      flash[:alert] = "There was an error pushing the item. (#{result[:status]}: #{result[:message]})"
+    end
+    redirect_to media_object_path(@media_object.id)
```

Comment:
The redirect is required to show the flash message?  I wish we could do this xhr and not have to reload the page just to show the flash but it might not be worth it.